main.tf

```diff
@@ -128,13 +128,13 @@ resource "aws_elasticache_subnet_group" "elasticache" {
 # NAT Gateway
 ##############
 resource "aws_eip" "nat" {
-  count = "${var.enable_nat_gateway ? (var.single_nat_gateway ? 1 : length(var.azs)) : 0}"
+  count = "${var.enable_nat_gateway ? (var.single_nat_gateway ? 1 : length(var.private_subnets)) : 0}"
```

Comment:
Such repetitive expressions you can simplify using locals:
```hcl
locals {
  number_of_nats =  "${var.enable_nat_gateway ? (var.single_nat_gateway ? 1 : length(var.private_subnets)) : 0}"
}

resource "aws_eip" "nat" {
  count = "${local.number_of_nats}"
  ...
}
```

I am just not sure that all users of this module are using the version which supports `locals`.

Let's keep it for later.

Reply:
Interesting! Hadn't seen those before...

Reply:
Actually, we should not use `locals`, because it is not possible to specify the version of module downloaded from the registry and it will break code for everyone who is using an older version of Terraform and using github instead of registry to download modules.